staging/src/k8s.io/legacy-cloud-providers/gce/gcpcredential/metadata.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package gcp
+package gcpcredential
 
```

Comment:
RegisterCredentialProvider below seems like one of the primary things we're pulling from k8s.io/kubernetes/pkg/credentialprovider. That is not gcp specific so it does not belong in either gcp or legacy-cloud-providers. We may want to look at making it something like kubelet or cloud-provider.

Reply:
Agree... the credentialprovider interface and secret-based provider seems to belong to k8s.io/kubelet more than k8s.io/cloud-provider

I like the move of the cloud-specific implementations to k8s.io/legacy-cloud-providers. cc @andrewsykim for visibility/intersection with https://github.com/kubernetes/kubernetes/pull/94196